tests/e2e/tidbcluster/serial.go

```diff
@@ -589,6 +589,229 @@ var _ = ginkgo.Describe("[Serial]", func() {
+			framework.ExpectNoError(err, "failed to get tidb pods")
+
+			ginkgo.By("Set --selector=version=old for the default TiDB Operator")
+			ocfg.Selector = []string{"version=old"}
+			ocfg.Tag = cfg.OperatorTag
+			ocfg.Image = cfg.OperatorImage
+			oa.InstallCRDOrDie(ocfg)
+			oa.UpgradeOperatorOrDie(ocfg)
+			log.Logf("Upgrade operator with --set-string \"selector=version=old\"")
+
+			ginkgo.By("Upgrade TiDB of TidbCluster 1, wait for 1min, check that no rolling update occurs")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Spec.Version = utilimage.TiDBV4
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 1 to upgrade tidb version to %v", utilimage.TiDBV4)
+
+			err = wait.Poll(5*time.Second, 1*time.Minute, func() (done bool, err error) {
+				// confirm the tidb haven't been changed
+				changed, err := utilpod.PodsAreChanged(c, tidbPods)()
+				if err != nil {
+					log.Logf("ERROR: meet error during verify tidb pods, err:%v", err)
+					return true, nil
+				}
+				if changed {
+					return true, nil
+				}
+				log.Logf("confirm tidb pods haven't been changed this time")
+
+				return false, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect pd/tikv/tidb haven't been changed for 5 minutes")
+			log.Logf("Upgrade TidbCluster 1 with new version, but not rolling update.")
+
+			ginkgo.By("Set label version=old to TidbCluster 1, check that TiKV of TidbCluster 1 has been rolling updated")
+			err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+				tc.Labels = map[string]string{"version": "old"}
+				return nil
+			})
+			framework.ExpectNoError(err, "failed to update TidbCluster 1 to set lable version=old")
+
+			err = wait.Poll(5*time.Second, 1*time.Minute, func() (done bool, err error) {
+				// confirm the tidb haven't been changed
+				changed, err := utilpod.PodsAreChanged(c, tidbPods)()
+				if err != nil {
+					log.Logf("ERROR: meet error during verify tidb pods, err:%v", err)
+					return true, nil
+				}
+				if changed {
+					return true, nil
+				}
+				log.Logf("confirm tidb pods haven't been changed this time")
+
+				return false, nil
+			})
+			framework.ExpectEqual(err, wait.ErrWaitTimeout, "expect pd/tikv/tidb have been changed for 5 minutes")
+			log.Logf("Upgrade TidbCluster 1 with label version=old and pod has been rolling updated.")
+
+			ginkgo.By("Deploy TidbCluster 2 with label version=new")
+			tc2Name := "tidbcluster2"
+			tc2 := fixture.GetTidbCluster(ns, tc2Name, utilimage.TiDBV4Prev)
+			tc2.Spec.PD.Replicas = 1
+			tc2.Spec.TiKV.Replicas = 1
+			tc2.Spec.TiDB.Replicas = 1
+			tc2.Labels = map[string]string{"version": "new"}
+
+			err = genericCli.Create(context.TODO(), tc2)
+			framework.ExpectNoError(err, "Expected TiDB cluster created")
+			log.Logf("Finished deploying TidbCluster 2 with label version=new.")
+
+			ginkgo.By("wait for 1min and check that no Pod created")
```

Comment:
Where does this wait?

Reply:
err = wait.Poll(5*time.Second, 5*time.Minute, func() (created bool, err error) {
				// confirm the Tidbcluster 2 tidb Pod don't be created
				tidbPods, err = getPods(labels.SelectorFromSet(label.New().Instance(tc2Name).TiDB().Labels()).String())
				if err != nil {
					log.Logf("ERROR: meet error during get tidb pods, err:%v", err)
					return true, nil
				}
				if len(tidbPods) != 0 {
					return true, nil
				}
				log.Logf("confirm tidb pods have be created this time")
				return false, nil
			})
			framework.ExpectEqual(err, wait.ErrWaitTimeout, "confirm Tidbcluster 2 no pod be created")
			log.Logf("confirm Tidbcluster 2 no pod be created")